doc/rst/language/evolution.rst

```diff
@@ -14,6 +14,42 @@ The purpose of this flag is to identify portions of a program that use a
 language or library feature has recently changed meaning or which is
 expected to change meaning in the future.
 
+version 1.34, March 2024
+------------------------
+
+.. _readme-evolution.default-task-intent-arrays:
+
+Default task intents for arrays
+*******************************
+
+In version 1.34, arrays have a slightly more relaxed default task intent. The
+default task intent for arrays was ``const`` by default since 1.32. This meant
+that to modify an array inside of a task parallel construct (``coforall``,
+``begin``, and ``cobegin``) always required a with clause to make the array
+modifiable. This change was only made to the task parallel constructs in 1.32,
+while ``forall`` was left as an unstable special case. 1.34 makes ``forall``,
+``coforall``, ``begin``, and ``cobegin`` consist with respect to their
+treatment of the default array intent.
+
+In 1.34, if an array is modifiable outside a parallel block, it is modifiable
+inside the parallel block.
```

Comment:
[I think I wrote this before, not seeing it now, so here it goes.]

I would like this section to start with a clear definition of the array intents in 1.34. Then, compare against 1.32. For example:

In version 1.34, the default task intent for an array variable is determined by whether the outer variable is "const". [.....] Therefore, if an array is modifiable outside a parallel block, it is modifiable inside the parallel block as well. It is no longer necessary to apply a with clause like `with (ref myArray)` in order to modify an outer array within a parallel construct. Cf. such a with-clause was neede in 1.32. This applies to forall, coforall, begin, cobegin.

Reply:
I rewrote this section, can you take another look and make sure it looks good to you?

Reply:
All good, thanks!